### DISCUSSION TITLE
LDAP Confuse new users

### DISCUSSION THREAD

**Discussion by null6process** [Ylianst/MeshCentral#6578] [Issues]
Hello,
I have an issue with creating users via the LDAP protocol on an Active Directory.

I have been using the same server for several years:

    Ubuntu 20.04 LTS
    MongoDB 6.0
    LDAP AD Server 2022

All users created in the AD have the same ID in MeshCentral, which is:
user//0105000000000005150000003d49fd5a5304fd666258fdfdfd2d0000

However, older users do not have this problem.

Here are the events:
![122153](https://github.com/user-attachments/assets/d037f8f3-677a-4607-982c-c54f7a371da1)

Have you encountered the same issue and found a solution?

**Comment by si458** [comment]
by default we look for `objectSid` then `objectGUID` then `name` then `cn` in that order
so chances are one of those values is duplicated across all your users (this happens with windows sometimes)
so you need to specify `ldapUserKey` to get a unique string value 
or use `ldapUserBinaryKey` to get a unique string value and convert it into hex (this looks like the option you are using)
and i believe the value should be set to `sAMAccountName` but remember changing this might break existing users permissions as there unique identifier in meshcentral might change!
so check before hand!

**Reply by null6process** [reply]
Thank you for your quick response. 
I exported the AD data to check objectSid, objectGUID, name, and cn.
The fields are different except for cn, which is empty for all my AD users.

ldapUserBinaryKey = objectSid

I am attaching the query result and the LDAP configuration file just in case.
Export users
```
	CN Name           ObjectGUID                           ObjectSid                                      
-- ----           ----------                           ---------                                      
   test test      094dxxxx-xxx7-xxxx-b1b7-xxxxxxc35bef         S-1-5-21-15xxxxx701-17xxxxxx15-367xxxx698-11762
   MxxxxxxxxxxxxU c5a9xxxx-xxx1-xxxx-b1a0-xxxxxx5a25a4          S-1-5-21-15xxxxx701-17xxxxxx15-367xxxx698-11741
   SxxxxxxxxxxxxT 93ccxxxx-xxx4-xxxx-a496-xxxxxx634193          S-1-5-21-15xxxxx701-17xxxxxx15-367xxxx698-11761
   oxxxxxxxxxx    0070xxxx-xxxa-xxxx-b44f-xxxxxx7996fa          S-1-5-21-15xxxxx701-17xxxxxx15-367xxxx698-1117 
```

Config File on server meshcentral: 

```
 domains": {
     "": {
      "_title": "MeshCentral 2",
       "title2": "Serveur <<DOMAIN NAME.COM>>",
 	  "auth": "ldap",
       "ldapUserName": "sAMAccountName",
       "ldapUserBinaryKey": "objectSid",
       "ldapOptions": {
         "url": "ldap://<<IP SERVER LDAP>>:389",
         "bindDN": "CN=<<USER LINK AD>>,DC=<<DOMAIN AD NAME>>,DC=local",
         "bindCredentials": "<<PASSWORD>>",
         "searchBase": "DC=<<DOMAIN AD NAME>>,DC=local",
         "searchFilter": "(&(objectCategory=Person)(sAMAccountName={{username}})(|(memberOf=cn=<<GROUP AD>>,DC=<<DOMAIN AD NAME>>,DC=local)(memberOf=cn=heart,ou=users,dc=company,dc=com)))"
       },
       "_minify": true,
       "_newAccounts": true,
 	  "newAccountsRights": [ "nonewgroups", "notools", "locked" ],
       "_userNameIsEmail": true
     }
```
In order to check the field matches,
Do you know how to convert the value of the ObjectSid field to obtain the ldapUserBinaryKey value?

I suspect there might be a variable overflow issue.
